models/articles.model.js

```diff
@@ -1,16 +1,73 @@
 const db = require("../db/connection");
 
-exports.getArticlesModel = () => {
+exports.getArticlesModel = (sort_by = "date", order = "desc", topic) => {
   let SQL = `SELECT articles.author, title, articles.article_id, 
   topic, articles.created_at, articles.votes, 
   COUNT(comments.article_id):: INTEGER AS comment_count
   FROM articles
-  LEFT JOIN comments ON articles.article_id = comments.article_id
-  GROUP BY articles.article_id
-  ORDER BY articles.created_at DESC
-  ;
+  LEFT JOIN comments ON articles.article_id = comments.article_id 
   `;
-  return db.query(SQL).then(({ rows }) => rows);
+
+  const queryValues = [];
+
+  const validSortBy = [
+    "article_id",
+    "author",
+    "title",
+    "topic",
+    "date",
+    "votes",
+    "comment_count",
+  ];
+  const validOrder = ["asc", "desc"];
+
+  if (!validSortBy.includes(sort_by) || !validOrder.includes(order)) {
+    return Promise.reject({ status: 400, msg: "Bad Request" });
+  }
+
+  if (topic !== undefined) {
+    SQL += ` WHERE topic = $1 GROUP BY articles.article_id`;
+    queryValues.push(topic);
+  } else {
+    SQL += ` GROUP BY articles.article_id`;
+  }
+
+  if (sort_by === "article_id") {
+    SQL += ` ORDER BY articles.article_id`;
+  } else if (sort_by === "author") {
+    SQL += ` ORDER BY articles.author`;
+  } else if (sort_by === "title") {
```

Comment:
this is not very DRY - if you have confirmed that the `sort_by` and `order` are in the array of approved values - then you can use string interpolation here 

i.e. - `SQL += ORDER BY articles.${sort_by}`